src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -321,13 +321,21 @@ private void ApplySettings(ConsumerSettings<K, V> updatedSettings)
                 if (_log.IsDebugEnabled)
                     _log.Debug($"Creating Kafka consumer with settings: {JsonConvert.SerializeObject(_settings)}");
 
+                var localSelf = Self;
                 _consumer = _settings.CreateKafkaConsumer(
-                    consumeErrorHandler: (c, e) => ProcessError(new KafkaException(e)),
-                    partitionAssignedHandler: (c, tp) => Self.Tell(new PartitionAssigned(tp.ToImmutableHashSet())),
-                    partitionRevokedHandler: (c, tp) => Self.Tell(new PartitionRevoked(tp.ToImmutableHashSet())),
+                    consumeErrorHandler: (c, e) =>
+                    {
+                        var exception = new KafkaException(e);
+                        ProcessError(exception);
+                        _pollCancellation?.Cancel();
+                        _log.Error(exception, "Exception when polling from consumer, stopping actor: {0}", exception.ToString());
+                        Context.Stop(Self);
```

Comment:
It would have to be a pretty severe error from the consumer to trigger this, right? Something like our idle connections being reaped would not cause this - correct?

Reply:
Incorrect, all errors are being processed, the actor kills itself after sending the failure message to the stage to be processed.
It is the job of the stage to decide if it needs to stop itself or restart the dead actor.

We keep on polling for messages even when there are no requests from downstream, this is done safely by pausing all subscribed topics before we do the poll calls, this ensures that only system messages are being processed by these blank polls.

The consequence is that these errors, while it usually happened during consume in normal kafka applications, will appear as if it happened "out of band", and some errors like the "all brokers are down" errors would not be propagated downstream, since there will be no requests pending, which is quite safe.

But we do need to make a custom decider on the stage itself that blocks these kind of nonsense "errors" and not have the stream crash and burn while there are no messages on the queue.

Reply:
> But we do need to make a custom decider on the stage itself that blocks these kind of nonsense "errors" and not have the stream crash and burn while there are no messages on the queue.

Agree